File: CS/PS/Site/programmers/lv1/모의고사.js
function solution(answers) {
  var answer = [];
  const a1 = [1, 2, 3, 4, 5];
  const a2 = [2, 1, 2, 3, 2, 4, 2, 5];
  const a3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];

  let countAnswer = Array(3).fill(0);
  let index = 0;
  for (let answer of answers){
    if(answer == a1[index % 5]){ countAnswer[0]++;}
    if(answer == a2[index % 8]){ countAnswer[1]++;}
    if(answer == a3[index % 10]){ countAnswer[2]++;}
    index++;
  }
  let maxValue = Math.max(...countAnswer);
  countAnswer.forEach((value, idx) => {
    if(value == maxValue)
    answer.push(idx + 1);
  });
    
  return answer;
}

console.log(solution([1,3,2,4,2]));

File: CS/PS/복습/progr_lv1/모의고사.py
# 18:03 ~ 18:10
def solution(answers):
    answer = []
    one = [1,2,3,4,5] # 5
    two = [2,1,2,3,2,4,2,5] # 8
    three = [3,3,1,1,2,2,4,4,5,5] # 10
    
    cnt = [0,0,0]
    for idx,v in enumerate(answers):
        if v == one[idx % 5]:
            cnt[0] += 1
        if v == two[idx % 8]:
            cnt[1] += 1
        if v == three[idx % 10]:
            cnt[2] += 1
    max_val = max(cnt)
    for idx, c in enumerate(cnt):
        if max_val == c:
            answer.append(idx + 1)
    return answer